repl/src/main/scala/com/cloudera/livy/repl/Main.scala

```diff
@@ -135,17 +133,17 @@ class ScalatraBootstrap extends LifeCycle with Logging {
 
         // Wait for our url to be discovered.
         val replUrl = waitForReplUrl()
-
-        var req = url(callbackUrl).setContentType("application/json", "UTF-8")
-        req = req << write(Map("url" -> replUrl))
-
         info(s"Calling $callbackUrl...")
-        val rep = Http(req OK as.String)
-        rep.onFailure {
-          case _ => System.exit(1)
+        val response = new AsyncHttpClient().preparePost(callbackUrl)
```

Comment:
Shouldn't you keep the reference to the new http client and close it in a finally block?